tools/dimspy/merge_peaklists.xml

```diff
@@ -10,52 +10,60 @@
         #for $fn in $hdf5_files_in
             --input '$fn'
         #end for
-        #if $multilist.flag == 'true'
+
+        #if $multilist
             --output .
-            --filelist $multilist.filelist'
         #else
             --output '$hdf5_file_out'
-            #if $filelist
-                --filelist '$multilist.filelist'
-            #end if
         #end if
+
+        #if $filelist
+           --filelist '$filelist'
+        #end if
+
     ]]>
     </command>
     <inputs>
-        <param name="hdf5_files_in" argument="--input" type="data" format="h5" multiple="true" min="2" label="Peaklist to merge" help="Select a minimum of two hdf5 files" />
-        <conditional name="multilist">
-            <param name="flag" type="boolean" checked="false" truevalue="true" falsevalue="false" label="Create multiple merged peaklists?" help="Based on a 'multilist' column in the filelist text file, divide the peaklists into different lists
-                                                       e.g. [pl1, pl2, pl3, pl4] can be converted to 2 new lists:
+        <param name="hdf5_files_in" type="data" multiple="true" format="h5" min="2" label="Peaklist to merge" help="Select a minimum of two hdf5 files" argument="--input" />
+        <param name="multilist" type="boolean" checked="false" label="Create multiple merged peaklists?"
+               help="Based on a 'multilist' column in the filelist text file, divide the peaklists into different lists
+                                                       e.g. [pl1 pl2], [pl3, pl4] can be converted to 2 new lists:
                                                             [pl1, pl3] [pl2, pl4]"/>
-            <when value="true">
-                <param name="filelist" argument="--filelist" type="data" format="tsv,tabular" label="Filelist" />
-            </when>
-            <when value="false">
-                <param name="filelist" argument="--filelist" type="data" format="tsv,tabular" optional="true" label="Filelist" />
-            </when>
-        </conditional>
+        <param name="filelist" type="data" format="tsv,tabular" optional="true" label="Filelist" argument="--filelist"  />
```

Comment:
if argumen is --filelist you can remove the `name` I would also add some more help to this parameter I guess

Reply:
I have kept the name for consistency with other params (e.g. hdf5_files_in and multilist). I have added a short help section.